pkg/devicetwin/devicetwin_test.go

```diff
@@ -0,0 +1,148 @@
+package devicetwin
+
+import (
+	"github.com/kubeedge/kubeedge/beehive/pkg/common/config"
+	"github.com/kubeedge/kubeedge/pkg/devicetwin/dtcontext"
+	"github.com/kubeedge/kubeedge/pkg/devicetwin/dtmodule"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+
+	"github.com/kubeedge/kubeedge/beehive/pkg/common/log"
+	"github.com/kubeedge/kubeedge/beehive/pkg/core"
+	"github.com/kubeedge/kubeedge/beehive/pkg/core/context"
+	"github.com/kubeedge/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/mocks/beego"
+	"github.com/kubeedge/kubeedge/mocks/beehive"
+	"github.com/kubeedge/kubeedge/pkg/common/dbm"
+)
+
+const (
+	//TestModule is name of test.
+	TestModule = "test"
+	//GroupResource is name of resource group resource.
+	GroupResource = "resource"
+)
+
+// ormerMock is mocked Ormer implementation.
+var ormerMock *beego.MockOrmer
+
+// querySeterMock is mocked QurySeter implementation.
+var querySeterMock *beego.MockQuerySeter
+
+// fakeDeviceTwin is mocked implementation of devicetwin module.
+var fakeDeviceTwin *beehive.MockModule
+
+// mainContext is beehive context used for communication between modules.
+var mainContext *context.Context
+
+// initMocks is function to initialize mocks.
+func initMocks(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	ormerMock = beego.NewMockOrmer(mockCtrl)
+	querySeterMock = beego.NewMockQuerySeter(mockCtrl)
+	fakeDeviceTwin = beehive.NewMockModule(mockCtrl)
+	dbm.DBAccess = ormerMock
+}
+
+// registerFakeModule is fuction to register all mock modules used.
+func registerFakeModule() {
+	fakeDeviceTwin.EXPECT().Name().Return(TestModule).Times(3)
+	core.Register(fakeDeviceTwin)
+	mainContext = context.GetContext(context.MsgCtxTypeChannel)
+	mainContext.AddModule(TestModule)
```

Comment:
what is the purpose of this TestModule ?

Reply:
This is to try sending a message from devicetwin to a module so as to check context is initialized properly.